### ISSUE TITLE
Nan Loss when training

### ISSUE THREAD

**Issue by fernando234234** [kohya-ss/musubi-tuner#6]
When i try to train using  the Tuner, i'm able to get all the way up to the training but i get a NANS loss, from the absolute start, 
I use a jpg (i tried a trasparent png) dataset as well, only images, no video, with .txt captions for each.
i used the preprocess_text_encoder_tokenizer_utils.py from the original repo.
This is the entire notebook:
[Untitled (2).md](https://github.com/user-attachments/files/18301658/Untitled.2.md)

Can anyone help me point out why the nans?

**Comment by sdbds** [comment]
I encountered a similar situation, it seemed to be caused by the official FP8.pt, I switched to the earliest model and then used --fp8 base without this problem

**Comment by mrT333** [comment]
I had the same problem. Only solution was to add `--split_attn `to the training command. 

I did two training runs. One I did few days ago without any problems. Another one I did today with this NaN Loss error. Interestingly I did the exact same training run (exactly the same dataset, same parameters etc., only difference was a lower learning rate). It confused me why I got the error today with essentially the same setup as few days ago which worked flawlessly.

The difference that caused the Nan error for me was that today the main branch contained a bugfix that made sure that split attention was actually turned off when it was not set to true.  This bugfix did not exist a few days ago, which means the "always on" split attention bug forced split attention on my training even without me having it turned on, but that was what prevented me from triggering the NaN loss bug a few days ago...

So there must be another bug lurking here independent of the cached latents or Text Encoder outputs, (I tried re-caching both with bf 16 and still got NaN), and that bug only gets triggered when split attention is turned off

**Comment by mrT333** [comment]
launching the script with the following parameters caused a NaN loss for me:

`accelerate launch --num_cpu_threads_per_process 1 --mixed_precision bf16 hv_train_network.py --dit HunyuanVideo/ckpts/hunyuan-video-t2v-720p/transformers/mp_rank_00_model_states.pt --dataset_config dataset/config_hunyuan_1.toml --sdpa --mixed_precision bf16 --optimizer_type adamw8bit --learning_rate 1e-4 --gradient_checkpointing --max_data_loader_n_workers 2 --persistent_data_loader_workers --network_module=networks.lora --network_dim=32 --timestep_sampling sigmoid --discrete_flow_shift 1.0 --max_train_epochs 4000 --save_every_n_epochs=50 --seed 42 --output_dir output_dir --output_name my_output_name`

simply adding `--split_attn` to the above script avoided the NaN loss trigger and the training worked as expected.

> Additionally, when using bf16 for caching, I observed that the training results collapsed. Therefore, bf16 caching doesn't appear to be a solution to this issue.

Do you mean the training results collapsed because of bf16 caching? Or it collapsed due to other reasons while using bf16 for caching? Should I expect my current training run - which is working fine currently -  to collapse as this time I also used bf16 for caching?

**Comment by viiika** [comment]
Met the same issue. Adding `--split_attn` to training script solved it.

**Comment by viiika** [comment]
Met the same issue again with deepspeed. remove --sdpa, use the --flash_attn then solve this again.
